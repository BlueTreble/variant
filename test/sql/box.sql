\set ECHO none
\set QUIET 1
/*
 * NOTE! Unlike the other cases of duplication, we need to handle numeric,
 * float and real separately because of odd things such as
 *
 * 0.0::float < 0.0::real = true
 */

\set baseline_type box
-- Space separated string
\set test_types 'box'
-- Used in array[ :base_values ]
\set base_values '$$((0,0),(0.5,0.5))$$, $$((0,0),(1,1))$$, $$((-1,-1),(1,1))$$, NULL'
\set compare_value ((0,0),(1,1))


\i test/helpers/type_setup.sql

/*
 * Box doesn't have sort operators, so do text-based sorting for the actual values.
 *
 * Note that this is used as a string value!
 */
\set op_test_order_by 'ORDER BY compare_type, base_type, compare_value::text COLLATE "C", base_value::text COLLATE "C"'

-- Box also doesn't have a != operator
DELETE FROM operator WHERE op ~ '!=';
DELETE FROM plan WHERE test_desc ~ '!= *$';

SELECT plan(sum(test_count)::int) FROM plan;

\i test/helpers/type.sql


-- vi: expandtab sw=2 ts=2
