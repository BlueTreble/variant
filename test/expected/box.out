\set ECHO 0




1..86
ok 1 - No NULL variants
ok 2 - Verify result of cast '(0.5,0.5),(0,0)'::box
ok 3 - Verify result of cast '(1,1),(0,0)'::box
ok 4 - Verify result of cast '(1,1),(-1,-1)'::box
ok 5 - Verify result of cast NULL::box
ok 6 - SET base_variant_b
ok 7 - Verify text_in(text_out()) works for (box,"(0.5,0.5),(0,0)")
ok 8 - Verify text_in(text_out()) works for (box,"(1,1),(0,0)")
ok 9 - Verify text_in(text_out()) works for (box,"(1,1),(-1,-1)")
ok 10 - Verify text_in(text_out()) works for (box,)
ok 11 - No NULL variants
ok 12 - Verify result of cast '(1,1),(0,0)'::box
ok 13 - SET compare_variant_b
ok 14 - Verify text_in(text_out()) works for (box,"(1,1),(0,0)")
psql:test/helpers/type.sql:135: ERROR:  operator does not exist: box <> box
LINE 2:       compare__casted !=  base__casted
                              ^
HINT:  No operator matches the given name and argument type(s). You might need to add explicit type casts.
QUERY:  SELECT is(
      compare__casted !=  base__casted
      , compare_value !=  base_value
      , format( 'check casted vs uncasted for %s !=  %s', compare_cast_string, base_cast_string )
    )
  FROM (
    SELECT * FROM op_test_data ORDER BY compare_type, base_type, compare_value::text, base_value::text
  ) ordered -- Need to order first otherwise our test numbers get re-arranged, which is confusing

CONTEXT:  PL/pgSQL function pg_temp_2.exec_text(text,anyarray) line 7 at FOR over EXECUTE statement
